Module 3 Edits- restore new Mod 2 (#193)
saucelabs/sauce-school: codelabs/Module3_APITesting.md

### Note:
Negative

> : In the previous module, [_Introduction to API Fortress_](https://training.saucelabs.com/codelabs/Module2-APITesting/index.html#0), we used the **Generate Test** button. This button automagically generates the HTTP request, assertions, and other test elements so that you can focus on the more intricate and tricky parts of your tests. In this module we will build a test from scratch to build understanding of the importance of assertions, but refer back to that test for inspiration and ideas on how to design your assertions.

Reviewer: ```suggestion
: In the previous module, [_Introduction to API Fortress_](https://training.saucelabs.com/codelabs/Module2-APITesting/index.html#0), we used the **Generate Test** button. This button automagically generates the HTTP request, assertions, and other test elements so that you can focus on the more intricate and tricky parts of your tests. In this module we will build a test from scratch in order to understand the importance of assertions, but refer back to that test for inspiration and ideas on how to design your assertions.
```
sounded wonky

1. This is what the finished result should look like:
   <img src="assets/apif-mod3/example_get_component.png" alt="Finished result of GET component" />

In the next section you will work with  **Assertion** components, learning how they are a vital part of your API tests.

<!-- ------------------------ -->

## 3.04 Assertion Component
Duration: 0:04:00

> There's a common phrase in the automated testing space: "If there is **no assertion**, it **isn't a test**." The previous sentence demonstrates a common [testing anti-pattern](http://www.everydayunittesting.com/2017/03/unit-testing-anti-pattern-not-asserting.html); even if your code doesn't throw an errors or crash, it doesn't mean it's a valid test.

Reviewer: ```suggestion
There's a common phrase in the automated testing space: "If there is **no assertion**, it **isn't a test**." The previous sentence demonstrates a common [testing anti-pattern](http://www.everydayunittesting.com/2017/03/unit-testing-anti-pattern-not-asserting.html); even if your code doesn't throw errors or crash, it doesn't mean it's a valid test.
```

grammar issue

   * Name: `domain`
   * Value: `saucelabs.com`
   <img src="assets/apif-mod3/domain-param.png" alt="Domain Parameter" />

Next we need to substitute the `domain` value in the current GET request:
1. In the menu to the left of the **GET** request component, select **Edit component**
   <img src="assets/apif-mod3/edit-component.png" alt="Edit Component" />
1. Substitute `saucelabs.com` with the `${domain}` global parameter:
   <img src="assets/apif-mod3/use-param.png" alt="Use Param" />
1. Run the test again to ensure the test still passes with the same result.

> 1. Create `protocol` and `uri` global parameters. The values should `https://` and `/versions.json` respectively

Reviewer: ```suggestion
1. Create `protocol` and `uri` global parameters. The values should be `https://` and `/versions.json` respectively
```

> 1. Add  Below is what the finished result should look like:

Reviewer: ```suggestion
1. Below is what the finished result should look like:
```

## 3.06 The Vault
Duration: 0:07:00

> The Vault is a unique feature of the API Fortress platform that allows you to store information for use across all projects. while Input Sets are typically only used within the same test or project, the vault allows you to store things that can be used across any project.

Reviewer: ```suggestion
The Vault is a unique feature of the API Fortress platform that allows you to store information for use across all projects. While Input Sets are typically only used within the same test or project, the vault allows you to store things that can be used across any project.
```

> The Vault allow you to save more than just variables, with the vault you can save, edit, and reuse almost anything including:

Reviewer: ```suggestion
The Vault allows you to save more than just variables, with the vault you can save, edit, and reuse almost anything including:
```

#### Note:
Negative

> : If you plan on re-using code snippets from the Vault, make sure those variables remain consistent across each test. Also note that if you set an input set or global parameter with the same name, those values will override what is saved in the Vault.

Reviewer: ```suggestion
: If you plan on re-using code snippets from the Vault, make sure those variables remain consistent across each test. Also note that if you add an input set or global parameter with the same name, those values will override what is saved in the Vault.
```
